cider/function/FunctionLookup.cpp

```diff
@@ -0,0 +1,222 @@
+            from_platform_));
+  }
+  const std::string& func_name = function_signature.func_name;
+  const auto& functionSignature =
+      cider::function::substrait::SubstraitFunctionSignature::of(
+          func_name, function_signature.arguments, function_signature.return_type);
+  const auto& functionOption =
+      aggregate_function_look_up_ptr_->lookupFunction(functionSignature);
+  if (functionOption.has_value()) {
+    return function_mappings_->getFunctionOpSupportType(func_name);
+  }
+  return OpSupportExprType::kUNDEFINED_EXPR;
+}
+
+const OpSupportExprType FunctionLookup::getExtensionFunctionOpSupportType(
+    const FunctionSignature& function_signature) const {
+  const PlatformType& from_platform = function_signature.from_platform;
+  if (from_platform != from_platform_) {
+    CIDER_THROW(
+        CiderCompileException,
+        fmt::format(
+            "Platform of target function is {}, mismatched with registered platform {}",
+            from_platform,
+            from_platform_));
+  }
+  const std::string& func_name = function_signature.func_name;
+  const auto& functionSignature =
+      cider::function::substrait::SubstraitFunctionSignature::of(
+          func_name, function_signature.arguments, function_signature.return_type);
+  const auto& functionOption =
+      extension_function_look_up_ptr_->lookupFunction(functionSignature);
+  if (functionOption.has_value()) {
+    return OpSupportExprType::kFUNCTION_OPER;
+  }
+  return OpSupportExprType::kUNDEFINED_EXPR;
+}
+
+/// first search extension function, second search internal function
+const OpSupportExprType FunctionLookup::getFunctionOpSupportType(
+    const FunctionSignature& function_signature) const {
+  OpSupportExprType result = OpSupportExprType::kUNDEFINED_EXPR;
+  result = getExtensionFunctionOpSupportType(function_signature);
+  if (result != OpSupportExprType::kUNDEFINED_EXPR) {
+    return result;
+  }
+  result = getScalarFunctionOpSupportType(function_signature);
+  if (result != OpSupportExprType::kUNDEFINED_EXPR) {
+    return result;
+  }
+  result = getAggFunctionOpSupportType(function_signature);
+  return result;
+}
+
+const FunctionDescriptor FunctionLookup::lookupFunction(
+    const FunctionSignature& function_signature) const {
+  FunctionDescriptor function_descriptor;
+  const PlatformType& from_platform = function_signature.from_platform;
+  if (from_platform != from_platform_) {
+    CIDER_THROW(
+        CiderCompileException,
+        fmt::format(
+            "Platform of target function is {}, mismatched with registered platform {}",
+            from_platform,
+            from_platform_));
+  }
+  function_descriptor.func_sig = function_signature;
+  function_descriptor.op_support_expr_type = getFunctionOpSupportType(function_signature);
+  if (function_descriptor.op_support_expr_type != OpSupportExprType::kUNDEFINED_EXPR &&
+      function_descriptor.op_support_expr_type == OpSupportExprType::kFUNCTION_OPER) {
```

Comment:
`if (function_descriptor.op_support_expr_type == OpSupportExprType::kFUNCTION_OPER)` should be enough.

Reply:
Done.